models/Task.php

```diff
@@ -1055,23 +1055,36 @@ public function getColor()
 
     // ###########  handle task related account  ###########
 
-    public function AddTaskAccount()
+    public function manageTaskAccount()
     {
-        $account = new Account([
-            'title' => "Task#$this->id ( $this->title )",
-            'space_id' => $this->content->container->id,
-            'account_type' => Account::TYPE_TASK,
-            'user_id' => empty($this->responsibleUsers) ? null : User::findOne(['guid' => $this->responsibleUsers[0]])->id
-        ]);
+        $accountTitle = "Task#$this->id ( $this->title )";
+        $accountUserId = empty($this->responsibleUsers) ? null : User::findOne(['guid' => $this->responsibleUsers[0]])->id;
 
-        $account->save();
+        if (null === ($taskAccount = TaskAccount::findOne(['task_id' => $this->id]))) {
 
-        $taskAccount = new TaskAccount([
-            'task_id' => $this->id,
-            'account_id' => $account->id,
-            'account_type' => self::ACCOUNT_SPACE,
-        ]);
+            $account = new Account([
+                'title' => $accountTitle,
+                'space_id' => $this->content->container->id,
+                'account_type' => Account::TYPE_TASK,
+                'user_id' => $accountUserId
+            ]);
+
+            $account->save();
+
+            $taskAccount = new TaskAccount([
+                'task_id' => $this->id,
+                'account_id' => $account->id,
+                'account_type' => self::ACCOUNT_SPACE,
+            ]);
+
+            return $taskAccount->save();
+        } else {
+            $account = Account::findOne(['id' => $taskAccount->account_id]);
+
+            $account->title = $accountTitle;
```

Comment:
@bumi, Here's the code that changes the account name when the task name is changed.

Reply:
oh.... guess I was blind :)